packages/transformers/webext/src/WebExtTransformer.js

```diff
@@ -0,0 +1,261 @@
+        dirname(filePath),
+        program.web_accessible_resources[i],
+      );
+      for (const fp of await glob(globQuery, fs, {})) {
+        asset.addURLDependency(relative(dirname(filePath), fp), {
+          isEntry: true,
+          loc: {
+            filePath,
+            ...getJSONSourceLocation(ptrs[`/web_accessible_resources/${i}`]),
+          },
+        });
+      }
+    }
+  }
+  for (let loc of DEP_LOCS) {
+    let isEntry: boolean = false;
+    if (typeof loc[loc.length - 1] == 'boolean') {
+      isEntry = ((loc[loc.length - 1]: any): boolean);
+      loc = loc.slice(0, -1);
+    }
+    const locStr = '/' + loc.join('/');
+    let obj: any = program;
+    for (let i = 0; i < loc.length - 1; ++i) {
+      obj = obj[loc[i]];
+      if (!obj) break;
+    }
+    if (!obj) continue;
+    const parent = obj,
+      lloc = loc[loc.length - 1];
+    obj = obj[lloc];
+    if (!obj) continue;
+    if (typeof obj == 'string')
+      parent[lloc] = asset.addURLDependency(
+        // TODO: not this, for sure
+        (extname(obj) == '.json' ? 'raw:' : '') + obj,
+        {
+          isEntry,
+          loc: {
+            filePath,
+            ...getJSONSourceLocation(ptrs[locStr], 'value'),
+          },
+        },
+      );
+    else {
+      for (const k of Object.keys(obj)) {
+        obj[k] = asset.addURLDependency(
+          (extname(obj[k]) == '.json' ? 'raw:' : '') + obj[k],
+          {
+            isEntry,
+            loc: {
+              filePath,
+              ...getJSONSourceLocation(ptrs[locStr + '/' + k], 'value'),
+            },
+          },
+        );
+      }
+    }
+  }
+}
+
+function cspPatchHMR(policy: ?string) {
+  if (policy) {
+    const csp = parseCSP(policy);
+    policy = '';
+    if (!csp['script-src']) {
+      csp['script-src'] = ["'self'", "'unsafe-eval'", 'blob: filesystem:'];
+    }
+    if (!csp['object-src']) {
+      csp['object-src'] = ["'self' blob: filesystem:"];
+    }
+    if (!csp['script-src'].includes("'unsafe-eval'")) {
+      csp['script-src'].push("'unsafe-eval'");
+    }
+    for (const k in csp) {
+      policy += k + ' ' + csp[k].join(' ') + ';';
+    }
+    return policy;
+  } else {
+    return (
+      "script-src 'self' 'unsafe-eval' blob: filesystem:;" +
+      "object-src 'self' blob: filesystem:;"
+    );
+  }
+}
+
+export default (new Transformer({
+  async parse({asset}) {
+    const code = await asset.getCode();
+    const map = parse(code);
+    if (BASE_KEYS.some(key => !map.data.hasOwnProperty(key))) {
+      // This is probably just another file that happens to be named manifest.json
+      return null;
+    }
+    validateSchema.diagnostic(
+      WebExtSchema,
+      {
+        data: map.data,
+        source: code,
+        filePath: asset.filePath,
+      },
+      '@parcel/transformer-webext',
+      'Invalid Web Extension manifest',
+    );
+    return {
+      type: 'json-source-map',
+      version: '0.6.1',
+      program: map,
+    };
```

Comment:
Using `parse` doesn't have any benefit here, you can just move that code into `transform`.

Reply:
It allows us to use another JSON transformer if we have a file that happens to be named `manifest.json` (for example, a web app manifest).